src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -187,5 +187,33 @@ public static class TimeSeriesCatalog
             new SsaForecastingEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, inputColumnName, windowSize, seriesLength, trainSize,
                 horizon, isAdaptive, discountFactor, rankSelectionMethod, rank, maxRank, shouldStabilize, shouldMaintainInfo, maxGrowth, confidenceLowerBoundColumn,
                 confidenceUpperBoundColumn, confidenceLevel, variableHorizon);
+
+        /// <summary>
+        /// Create <see cref="SrCnnEntireAnomalyEstimator"/>, which detects timeseries anomalies for entire input using SRCNN algorithm.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.
+        /// The column data is a vector of <see cref="System.Double"/>. The length of this vector varies depending on <paramref name="srCnnDetectMode"/>.</param>
+        /// <param name="inputColumnName">Name of column to transform. The column data must be <see cref="SrCnnTsPoint"/>.</param>
+        /// <param name="threshold">The threshold to determine anomaly, score larger than the threshold is considered as anomaly. Must be in [0,1]. Default value is 0.3.</param>
```

Comment:
Why the restriction that the inputColumnName must be of type SrCnnTsPoint? The anomaly detectors we already ship do not have this restriction. 

Is it because of the need for a single structure that has both timestamp and the associated value to detect the anomaly on? 

If we have this restriction, how will this affect usage when data is loaded from file through a TextLoader? 

Would it make more sense to have a separate parameter in the API that identifies what the timestamp column is?

Reply:
I think it would be easier and more user friendly to make the input for this transformer be two columns (one of type `DateTime` and one of type `double`. I would even consider allowing the `Timestamp` input to be of type `long`, since `ML.NET`'s support for `DateTime` is very basic (it can only be loaded by `TextLoader` if the values are written in a very specific format, and there are currently no transformers that output `DateTime` type).

---
In reply to: [415369363](https://github.com/dotnet/machinelearning/pull/4998#discussion_r415369363) [](ancestors = 415369363)

Reply:
I agree with @yaeldekel's suggestion above. 

---
In reply to: [415894364](https://github.com/dotnet/machinelearning/pull/4998#discussion_r415894364) [](ancestors = 415894364,415369363)

Reply:
Thanks for your suggestions. It's no problem for us to split the SrCnnTsPoint into two columns, since this is not a very complex data structure. But I have some concerns here, to my understanding, TextLoader is something to make user easier to read input data from a file, in yahoo dataset and NAB dataset that PowerBI uses, the timestamps are represented in formatted datetime strings, if we make it long, the user still need to do some conversion and could not get dataview directly. Besides, the long timestamp also have different interpretings, is it Unix timestamp in seconds or in milliseconds, or in microseconds, etc? If TextLoader is a must supported way, could you consider to add some support for DateTime, since if ML.NET decides to support timeseries related models, the needs for DateTime type may be something common in the future.

Reply:
> I agree with @yaeldekel's suggestion above.
> 
> In reply to: [415894364](https://github.com/dotnet/machinelearning/pull/4998#discussion_r415894364) [](ancestors = 415894364,415369363)

Thanks for your suggestions. It's no problem for us to split the SrCnnTsPoint into two columns, since this is not a very complex data structure. But I have some concerns here, to my understanding, TextLoader is something to make user easier to read input data from a file, in yahoo dataset and NAB dataset that PowerBI uses, the timestamps are represented in formatted datetime strings, if we make it long, the user still need to do some conversion and could not get dataview directly. Besides, the long timestamp also have different interpretings, is it Unix timestamp in seconds or in milliseconds, or in microseconds, etc? If TextLoader is a must supported way, could you consider to add some support for DateTime, since if ML.NET decides to support timeseries related models, the needs for DateTime type may be something common in the future.

Reply:
TextLoader does support loading DateTime objects from text files, and in case there's ambiguity on how to load the format in the file, then users can use a CustomMappingTransformer, as shown in the tests I've recently added to DateTimeTransformerTests in this PR:

https://github.com/dotnet/machinelearning/pull/5063/files

Reply:
> I agree with @yaeldekel's suggestion above.
> 
> In reply to: [415894364](https://github.com/dotnet/machinelearning/pull/4998#discussion_r415894364) [](ancestors = 415894364,415369363)
Fixed.